bodhi/tests/server/consumers/test_masher.py

```diff
@@ -1147,6 +1148,166 @@ def test_obsolete_older_updates(self, publish, *args):
+            'completion_ts': 1187997970,
+            'creation_event_id': 151517,
+            'creation_time': '2007-08-24 19:38:29.422344',
+            'epoch': None,
+            'extra': None,
+            'id': 16058,
+            'name': 'TurboGears',
+            'nvr': 'TurboGears-1.0.2.2-2.fc17',
+            'owner_id': 388,
+            'owner_name': 'lmacken',
+            'package_id': 8,
+            'package_name': 'TurboGears',
+            'release': '2.fc17',
+            'state': 1,
+            'tag_id': 19,
+            'tag_name': 'f17-updates-candidate',
+            'task_id': 127621,
+            'version': '1.0.2.2'
+        }
+
+        rpms = [
+            {
+                'arch': 'src',
+                'build_id': 6475,
+                'buildroot_id': 1883,
+                'buildtime': 1178868422,
+                'epoch': 1,
+                'id': 62330,
+                'name': 'TurboGears',
+                'nvr': 'TurboGears-1.0.2.2-2.fc17',
+                'payloadhash': '6787febe92434a9be2a8f309d0e2014e',
+                'release': '2.fc17',
+                'size': 761742,
+                'version': '1.0.2.2'
+            },
+            {
+                'arch': 'noarch',
+                'build_id': 6475,
+                'buildroot_id': 1883,
+                'buildtime': 1178868537,
+                'epoch': 1,
+                'id': 62331,
+                'name': 'TurboGears',
+                'nvr': 'TurboGears-1.0.2.2-2.fc17',
+                'payloadhash': 'f3ec9bdce453816f94283a15a47cb952',
+                'release': '2.fc17',
+                'size': 1993385,
+                'version': '1.0.2.2'
+            }
+        ]
+
+        builds_md = []
+        for i, build in enumerate(msg["body"]["msg"]["updates"]):
+            nsv_list = build.split("-")
+            build_name = nsv_list[0]
+            build_version = nsv_list[2]
+            prot_build = template_build.copy()
+            prot_build["build_id"] += i
+            prot_build["creation_event_id"] += i
+            prot_build["nvr"] = build
+            prot_build["id"] += i
+            prot_build["name"] = build_name
+            prot_build["package_id"] += i
+            prot_build["package_name"] = build_name
+            prot_build["release"] = "f27"
+            prot_build["tag_name"] = "f27-modular-updates"
+            prot_build["task_id"] += i
+            prot_build["version"] = build_version
+            builds_md.append(prot_build)
+
+        listtagged.return_value = builds_md
+        listbuildrpms.return_value = rpms
+
+        self.masher.consume(msg)
+
+        publish.assert_called_with(topic="mashtask.complete",
+                                   force=True,
+                                   msg=dict(success=True,
+                                            repo='f27-modular-updates',
+                                            agent='mcurlej'))
```

Comment:
This test only makes this one assertion - it doesn't assert that the other mocked items were called correctly, or that other side effects happened as expected. We need much stronger assertions that things worked properly here.

Reply:
Cool - agreed in video chat that we'll put a few more assertions here (but no need to assert on *every* mock).

Reply:
done, added assertion for mocks called, and also added logic which checks if the mash has updated repomd.xml with updateinfo and also if updateinfo was created on disk. Running out of things to test here as its mostly mocked. So if you want more, give me suggestions... :) (most of it is the same code as for the old masher anyways.)